src/main/java/com/ksm/wstbackoffice/controller/CountryController.java

```diff
@@ -44,4 +49,20 @@ public CountryController(CountryService countryService) {
         return ResponseEntity.status(HttpStatus.OK)
                 .body(country);
     }
+
+    @PostMapping("/filter-name")
+    public ResponseEntity<Map<String, Map<String, ?>>> findCountriesStartWith(@RequestBody Collection<String> nameStartsWithFilters) {
+        if (nameStartsWithFilters.isEmpty()) {
+            return ResponseEntity.status(HttpStatus.BAD_REQUEST).body(null);
+        }
+
+        Map<String, Map<String, ?>> filteredCountries = countryService.startWith(nameStartsWithFilters);
```

Comment:
Please change
countryService.startWith(nameStartsWithFilters);
to
countryService.findAllBy(nameStartsWithFilters);

Reply:
done